lib/ex_doc/formatter/epub.ex

```diff
@@ -50,6 +50,23 @@ defmodule ExDoc.Formatter.EPUB do
     Path.relative_to_cwd(epub)
   end
 
+  @doc """
+  Helper that replaces anchor names and links that could potentially cause problems on EPUB documents
+
+  This helper replaces all the `&` with `&amp;` found in anchors like
+  `Kernel.xhtml#&&/2` or `<h2 id="&&&/2-examples" class="section-heading">...</h2>`
+
+  These anchor names cause a fatal error while EPUB readers parse the files,
+  resulting in truncated content.
+
+  For more details, see: https://github.com/elixir-lang/ex_doc/issues/1851
+  """
+  def fix_anchors(content) do
+    content
+    |> String.replace(~r{id="&+/\d+[^"]*}, &String.replace(&1, "&", "&amp;"))
+    |> String.replace(~r{href="[^#"]*#&+/\d+[^"]*}, &String.replace(&1, "&", "&amp;"))
```

Comment:
I frowned a little with these nested `String.replace`. So, please let me now if you have any advice on how to improve this function.

Reply:
@wojtekmach I though we had already escaped those when generating the links. Maybe this is something (or an option) we can pass when autolinking? The id we can fix by escaping in the document itself.

Reply:
@wojtekmach @josevalim I know it has been a while, but is there any action that's expected on my end? How do we move on/resume this discussion?